### PR TITLE
Release/5.7.0

### PR DESCRIPTION
**Pull request by jjordan-quantum**
### Added
- Support for signed numbers
- Passport interface, ABI, and Structs

### Changed
- Updated RPCSelector to throw when no healthy RPC is found

### CODE REVIEW

**Review by ramos-ph — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Changed
+- Updated RPCSelector Timestamp strategy to not take too much time to run
+
```
Comment (ramos-ph) [inline CHANGELOG.md:11]: Please rebase this with `main`. The changelog is different.

File: test/services/LeafValueCoder.test.ts [hunk 2]
```diff
@@ -149,5 +153,35 @@ describe('LeafValueCoder', () => {
         LeafValueCoder.printableValue('0x0000000000000000000000000000000000000000000000000000000000000000', 'TEST')
       ).to.eql('0');
     });
+
+    it('converts the value of a signed value', () => {
```
Comment (DZariusz) [inline test/services/LeafValueCoder.test.ts:157]: please merge all this tests into loop

Comment (jjordan-quantum) [reply test/services/LeafValueCoder.test.ts]: yep!

File: package.json [hunk 3]
```diff
@@ -42,6 +45,9 @@
     "cpx": "~1.5.0",
     "dotenv": "~8.2.0",
     "eslint": "~7.15.0",
+    "ethereum-waffle": "~3.4.0",
+    "ethers": "~5.4.5",
```
Comment (ramos-ph) [inline package.json:49]: Ethers is duplicated. Remove this line

Comment (jjordan-quantum) [reply package.json]: ok

**Review by ramos-ph — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -3,3 +3,8 @@ node_modules/
 dist/
 .env*
 .vscode
+.DS_STORE
+.idea
+cache
+src/contracts/examples/artifacts/
```
[the remark this answers is not in the document]
Comment (jjordan-quantum) [reply .gitignore:9]: i was instructed to ignore artifacts
they are produced when testing the example contract

**Review by matheushahnn — APPROVED** [review]

**Review by DZariusz — APPROVED** [review]

**Review by dougmrqs — APPROVED** [review]

**Pull request merged**